# 問題1

(a) 線形独立  

(b) $\left(\begin{array}{ccc}0.4 \\ -0.3 \end{array}\right)$のノルムが1でないため，正規直交ではなく単に直交するだけ  

(c) ノルムも1であるため，正規直交

# 問題2

(a) $Q = \left(
    \begin{array}{ccc}
        1 & 0 \\
        0 & 0 \\
        0 & 1
    \end{array}
  \right)$  

(b) 任意のベクトルとゼロベクトルのペア  

(c) $A = \left(
    \begin{array}{ccc}
        1/2 & 1/2 & 1/2 & 1/2 \\
        1/2 & 1/2 & -1/2 & -1/2 \\
        1/2 & -1/2 & 1/2 & -1/2 \\
        1/2 & -1/2 & -1/2 & 1/2
    \end{array}
  \right)$  

# 問題3

転置行列であるため，$Q_1 Q_1^t = E$, $Q_2 Q_2^t = E$ である．

そこで$(Q_1Q_2)(Q_1Q_2)^t$を考えると，  

\begin{eqnarray*}
(Q_1Q_2)(Q_1Q_2)^t&=&(Q_1Q_2)(Q_2^tQ_1^t)\\
 &=&Q_1(Q_2Q_2^t)Q_1^t\\
 &=&Q_1Q_1^t\\
 &=& E
\end{eqnarray*}

ゆえに$Q_1Q_2$も直交行列である．

# 問題4
正規直交するベクトル群 $q_1, q_2, q_3$の話

(a)ベクトルによる証明  

$c_1q_1+c_2q_2+c_3q_3=0$である．  
ここで，ベクトル $c_1q_1+c_2q_2+c_3q_3$とベクトル $q_1$の内積を考える．  

$q_i \cdot q_i = 1$，$q_i \cdot q_j = 0 (i \neq j)$という正規直交ベクトルの性質より，  

$c_1 = 0$.

同様に$c_2 = 0, c_3 = 0$


(b)行列による証明

$c_1q_1+c_2q_2+c_3q_3=0$を行列を用いて書き直すと，  
$Q = \left(
    \begin{array}{ccc}
        q_1& q_2 & q_3 \\
    \end{array}
  \right)$，
$x = \left(
    \begin{array}{ccc}
        c_1 \\
        c_2 \\
        c_3
    \end{array}
  \right)$において$Qx = 0$と表す事ができる．  
  
$Q$は正規直交行列であるため，$Q^tQ = E$という性質を持つ．  

そこで$Qx=0$に$Q^t$をかけて

\begin{eqnarray*}
Q^t Qx&=&Q^t 0\\
     x&=& 0
\end{eqnarray*}

# 問題5

# 問題6
恐らく問題文の誤植で  
図1.7 → 図1.19だと思われます．

$A^tA = \left(
    \begin{array}{ccc}
        1 & 1 \\
        1 & 2
    \end{array}
  \right)$  

In [8]:
import numpy as np
A = np.array([[1, 1], [1, 2]])
la, v = np.linalg.eig(A)
print(la)

[ 0.38196601  2.61803399]


すなわち，固有値は0.38196601と2.61803399であり，  

$\lambda_{max} = 2.61803399$である．  

ここで$\frac{1}{2}(3+\sqrt{5})$を調べると，

In [10]:
(3+np.sqrt(5))/2

2.6180339887498949

よって$\lambda_{max} = \frac{1}{2}(3+\sqrt{5})$であることがわかる．  
また

\begin{eqnarray*}
\frac{1}{2}(3+\sqrt{5}) &=& \frac{1}{4}(6+2\sqrt{5}) \\
                        &=& (\frac{1}{2}(1+\sqrt{5}))^2
\end{eqnarray*}

よって，$\|A\| = \frac{1}{2}(1+\sqrt{5})$

# 問題7

$A = \left(
    \begin{array}{ccc}
        3 & 0 \\
        0 & 2
    \end{array}
  \right)$  

In [12]:
A = np.array([[3, 0], [0, 2]])
la, v = np.linalg.eig(A)
print(la)

[ 3.  2.]


よって  
$\lambda_{max} = 3$  

$\lambda_{max}/\lambda_{min} = 3/2 = 1.5$

$A = \left(
    \begin{array}{ccc}
        2 & 1 \\
        1 & 2
    \end{array}
  \right)$  

In [18]:
A = np.array([[2, 1], [1, 2]])
la, v = np.linalg.eig(A)
print(la)

[ 3.  1.]


よって  
$\lambda_{max} = 3$  

$\lambda_{max}/\lambda_{min} = 3/1 = 3$

$A = \left(
    \begin{array}{ccc}
        3 & 1 \\
        1 & 1
    \end{array}
  \right)$  

In [17]:
A = np.array([[3, 1], [1, 1]])
la, v = np.linalg.eig(A)
print(la)

[ 3.41421356  0.58578644]


よって  
$\lambda_{max} = 3.41421356$  

$\lambda_{max}/\lambda_{min} = 3.41421356/0.58578644 = 5.828427097$

# 問題8
たぶん与えられた行列が$A$であり，まず$A^tA$の計算から行うんだと思います．

(a)

$A = \left(
    \begin{array}{ccc}
        1 & 7 \\
        1 & 1
    \end{array}
  \right)$  

In [25]:
A = np.array([[1, 7], [1, 1]])
A_t = np.array([[1, 1], [7, 1]])

In [26]:
A_t_A = np.dot(A_t, A)
la, v = np.linalg.eig(A_t_A)
print(la)

[  0.70177872  51.29822128]


つまり，

$\|A\|^2 = 51.29822128$  

$\|A^{-1}\|^2 = 0.70177872$  

よって，

$\|A\| = 7.1622776600743423$

$\|A^{-1}\| = 0.83772234063560702$


ゆえに条件数 $c(A)$ は，

$c(A) = \frac{\|A\|}{\|A^{-1}\|}$ = 8.549703539

(b)

$A = \left(
    \begin{array}{ccc}
        1 & 1 \\
        0 & 0
    \end{array}
  \right)$  

In [29]:
A = np.array([[1, 1], [0, 0]])
A_t = np.array([[1, 0], [1, 0]])

In [30]:
A_t_A = np.dot(A_t, A)
la, v = np.linalg.eig(A_t_A)
print(la)

[ 2.  0.]


よって，

$\|A\| = \sqrt{2}$

$\|A^{-1}\| = 0$

この場合，条件数がどうなるのかわかりません．

(c)

$A = \left(
    \begin{array}{ccc}
        1 & 1 \\
        -1 & 1
    \end{array}
  \right)$  

In [31]:
A = np.array([[1, 1], [-1, 1]])
A_t = np.array([[1, -1], [1, 1]])

In [32]:
A_t_A = np.dot(A_t, A)
la, v = np.linalg.eig(A_t_A)
print(la)

[ 2.  2.]


よって，

$\|A\| = \sqrt{2}$

$\|A^{-1}\| = \sqrt{2}$

ゆえに条件数  

$c(A) = 1$